toolkit/tools/pkg/imagecustomizerlib/savedconfigs.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/file"
+)
+
+// 'SavedConfigs' is a subset of the Image Customizer input configurations that
+// needs to be saved on the output media so that it can be used in subsequent
+// runs of the Image Customizer against that same output media.
+//
+// This preservation of input configuration is necessary for subsequent runs if
+// the configuration does not result in updating root file system.
+//
+// For example, if the user specifies a kernel argument that is specific to the
+// ISO image, it will not be present in any of the grub config files on the
+// root file system - only in the final ISO image grub.cfg. When that ISO image
+// is further customized, the root file system grub.cfg migith get re-generated
+// and we need to remember to add the ISO specific arguments from the previous
+// runs. SavedConfigs is the place where we can store such arguments so we can
+// re-apply them.
+type SavedConfigs struct {
+	Iso imagecustomizerapi.Iso `yaml:"iso"`
```

Comment:
sure - I'm a bit borderline here since re-using types also means reusing validation code and other helper functions that work on such types. So, we'd be sacrificing these on the long term. I've created a version of the iso config for saving. Let's see how this evolve over time.